src/AppInstallerRepositoryCore/Microsoft/Schema/Portable_1_0/PortableTable.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "PortableTable.h"
+#include "SQLiteStatementBuilder.h"
+#include "Microsoft/Schema/IPortableIndex.h"
+
+namespace AppInstaller::Repository::Microsoft::Schema::Portable_V1_0
+{
+    using namespace std::string_view_literals;
+    static constexpr std::string_view s_PortableTable_Table_Name = "portable"sv;
+    static constexpr std::string_view s_PortableTable_Index_Separator = "_"sv;
+    static constexpr std::string_view s_PortableTable_Index_Suffix = "_index"sv;
+
+    static constexpr std::string_view s_PortableTable_FilePath_Column = "filePath"sv;
+    static constexpr std::string_view s_PortableTable_FileType_Column = "fileType"sv;
+    static constexpr std::string_view s_PortableTable_SHA256_Column = "sha256"sv;
+    static constexpr std::string_view s_PortableTable_SymlinkTarget_Column = "symlinkTarget"sv;
+
+    std::string_view PortableTable::TableName()
+    {
+        return s_PortableTable_Table_Name;
+    }
+    
+    void PortableTable::Create(SQLite::Connection& connection)
+    {
+        using namespace SQLite::Builder;
+
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "createPortableTable_v1_0");
+
+        StatementBuilder createTableBuilder;
+        createTableBuilder.CreateTable(s_PortableTable_Table_Name).BeginColumns();
+
+        createTableBuilder.Column(ColumnBuilder(s_PortableTable_FilePath_Column, Type::Text).NotNull());
+        createTableBuilder.Column(ColumnBuilder(s_PortableTable_FileType_Column, Type::Int64).NotNull());
+        createTableBuilder.Column(ColumnBuilder(s_PortableTable_SHA256_Column, Type::Blob).NotNull());
+        createTableBuilder.Column(ColumnBuilder(s_PortableTable_SymlinkTarget_Column, Type::Text));
+
+        createTableBuilder.EndColumns();
+        createTableBuilder.Execute(connection);
+
+        savepoint.Commit();
+    }
+
+    std::optional<SQLite::rowid_t> PortableTable::SelectByFilePath(const SQLite::Connection& connection, const std::filesystem::path& path)
+    {
+        SQLite::Builder::StatementBuilder builder;
+        builder.Select(SQLite::RowIDName).From(s_PortableTable_Table_Name).Where(s_PortableTable_FilePath_Column);
+        builder.Equals(path.u8string());

```

Comment:
>Equals [](http://example.com/codeflow?start=16&length=6)

Shall we do case insensitive for comparing paths?

Reply:
I left as is. Input for filepath is already normalized from the SetFilePath() function which calls std::filesystem::weakly_canonical() prior to assigning the member value. This ensures that the input is in normal form.

Reply:
Added extension method "COLLATE NOCASE" to column builder as well as tests to verify that column is case-insensitive.